linux/library/include/flutter_desktop_embedding/plugin.h

```diff
@@ -31,14 +36,24 @@ class Plugin {
   // |input_blocking| Determines whether user input should be blocked during the
   // duration of this plugin's platform callback handler (in most cases this
   // can be set to false).
-  explicit Plugin(std::string channel, bool input_blocking = false)
-      : channel_(channel), input_blocking_(input_blocking) {}
+  explicit Plugin(std::string channel, PlatformCallback platform_callback,
```

Comment:
I'm confused about what this means.  The plugin can't make a call to the engine unless it has an instance of it.  Should we then be including a field with a pointer to the flutter engine along with the base class?

Reply:
Updated this to take a setter method.  Unsure if this is what you had in mind.

Reply:
The client is still responsible providing the callback, not just the engine as discussed above. Summarizing from the offline discussion earlier, my take on this is:
- The callback logic is always going to be the same modulo the engine pointer.
- That logic is fundamentally about the details of the message format used by the engine.
- The core purpose of the Plugin class is to provide a higher-level abstraction around communicating with the engine.
Given that, a protected method within Plugin seems like the right place for that logic to live, rather than requiring something outside of Plugin to implement it and provide that implementation.

Reply:
Ping on this. I accidentally replied as a separate comment rather than as part of the review, so I'm guessing it was lost.